eval/eval.go

```diff
@@ -81,6 +136,13 @@ func (s *State) evalAssignment(right object.Object, node *ast.InfixExpression) o
 	}
 	switch node.Left.Value().Type() {
 	case token.DOT, token.LBRACKET:
+		nodeType := node.Type()
+		if isCompound(nodeType) {
+			opToEval := nodeType - (token.SUMASSIGN - token.PLUS)
+			res, _ := s.compoundAssignNested(node.Left, opToEval, right) // no need to check error, res is returned either way.
```

Comment:
that's likely wrong... it should be
```go
res, oerr := ...
if oerr != nil {
   return oerr
}
```
though the goal is to bubble up in error so maybe only needed in the recursion part 
but I guess in this case there is always something to calculate/return so just change the signature of compoundAssignNested to just return the result (which may be an error obj)